# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreParent.h
# Commit: 30d5dfedf252
# Full Hash: 30d5dfedf252f4f10a0d8948e86f4e57f1c4ccf9
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 8 - CookieStore API - e10s support, r=smaug,cookie-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217574
# ==============================================================================

diff -r fc3b4c5ee4b2 -r 30d5dfedf252 dom/cookiestore/CookieStoreParent.h
--- a/dom/cookiestore/CookieStoreParent.h	Tue Sep 10 16:31:40 2024 +0000
+++ b/dom/cookiestore/CookieStoreParent.h	Tue Sep 10 16:31:41 2024 +0000
@@ -12,7 +12,7 @@
 
 namespace mozilla::dom {
 
-class CookieStoreService;
+class CookieStoreNotificationWatcher;
 
 class CookieStoreParent final : public PCookieStoreParent {
   friend class PCookieStoreParent;
@@ -22,7 +22,7 @@
       MozPromise<CopyableTArray<CookieData>, nsresult, true>;
   using SetDeleteRequestPromise = MozPromise<bool, nsresult, true>;
 
-  NS_INLINE_DECL_REFCOUNTING(CookieStoreParent)
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(CookieStoreParent)
 
   CookieStoreParent();
 
@@ -38,14 +38,42 @@
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
       const nsString& aName, const nsString& aValue, const bool& aSession,
       const int64_t& aExpires, const nsString& aPath, const int32_t& aSameSite,
-      const bool& aPartitioned, SetRequestResolver&& aResolver);
+      const bool& aPartitioned, const nsID& aOperationID,
+      SetRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvDeleteRequest(
       const nsString& aDomain, const OriginAttributes& aOriginAttributes,
       const nsString& aName, const nsString& aPath, const bool& aPartitioned,
-      DeleteRequestResolver&& aResolver);
+      const nsID& aOperationID, DeleteRequestResolver&& aResolver);
 
   mozilla::ipc::IPCResult RecvClose();
+
+  void GetRequestOnMainThread(const nsAString& aDomain,
+                              const OriginAttributes& aOriginAttributes,
+                              bool aMatchName, const nsAString& aName,
+                              const nsACString& aPath, bool aOnlyFirstMatch,
+                              nsTArray<CookieData>& aResults);
+
+  // Returns true if a cookie notification has been generated while completing
+  // the operation.
+  bool SetRequestOnMainThread(const nsAString& aDomain,
+                              const OriginAttributes& aOriginAttributes,
+                              const nsAString& aName, const nsAString& aValue,
+                              bool aSession, int64_t aExpires,
+                              const nsAString& aPath, int32_t aSameSite,
+                              bool aPartitioned, const nsID& aOperationID);
+
+  // Returns true if a cookie notification has been generated while completing
+  // the operation.
+  bool DeleteRequestOnMainThread(const nsAString& aDomain,
+                                 const OriginAttributes& aOriginAttributes,
+                                 const nsAString& aName, const nsAString& aPath,
+                                 bool aPartitioned, const nsID& aOperationID);
+
+  CookieStoreNotificationWatcher* GetOrCreateNotificationWatcherOnMainThread(
+      const OriginAttributes& aOriginAttributes);
+
+  RefPtr<CookieStoreNotificationWatcher> mNotificationWatcherOnMainThread;
 };
 
 }  // namespace mozilla::dom